packages/js-api/src/shared/utils.ts

```diff
@@ -93,16 +93,28 @@ class CallQueue {
     })
   }
 
-  public hasIdenticalCall(call: string) {
+  public hasIdenticalRecentCall(callKey: string) {
     const now = new Date()
     this.queue = this.queue.filter((item) => now.getTime() - item.time < this.ttlInMS)
-    return this.queue.find((item) => item.call === call)
+    return this.queue.find((item) => item.call === callKey)
+  }
+
+  public hasCall(callKey: string) {
+    return this.queue.find((item) => item.call === callKey)
+  }
+
+  public empty() {
+    this.queue = []
```

Comment:
We could run through the queue and [abort](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort) pending requests to avoid potential race conditions from resolving empty responses below. Dealer's choice though, if you think it's safe as-is I don't have any religious objection to leaving it.